lgi/marshal.c

```diff
@@ -363,10 +364,9 @@ marshal_2c_array (lua_State *L, GITypeInfo *ti, GIArrayType atype,
 	  luaL_checktype (L, narg, LUA_TTABLE);
 
 	  /* Find out how long array should we allocate. */
-	  zero_terminated = g_type_info_is_zero_terminated (ti);
+	  zero_terminated = gi_type_info_is_zero_terminated (ti);
 	  objlen = lua_objlen (L, narg);
-	  *out_size = g_type_info_get_array_fixed_size (ti);
-	  if (atype != GI_ARRAY_TYPE_C || *out_size < 0)
+          if (atype != GI_ARRAY_TYPE_C || !gi_type_info_get_array_fixed_size (ti, (gsize *)&out_size))
```

Comment:
This change leads to a segfault when trying to construct objects using a table as the parameter. Referencing `out_size` here is unnecessary.

```suggestion
          if (atype != GI_ARRAY_TYPE_C || !gi_type_info_get_array_fixed_size (ti, (gsize *)out_size))
```